web-local/src/lib/components/workspace/explore/filters/Filters.svelte

```diff
@@ -42,6 +48,53 @@ The main feature-set component for dashboard filters
     return filters.include.length > 0 || filters.exclude.length > 0;
   }
 
+  let topListQuery;
+  let searchText = "";
+  let searchedValues = [];
+  let activeDimensionName;
+  let activeDimensionId;
+
+  $: addNull = "null".includes(searchText);
+
+  $: if (activeDimensionName && activeDimensionId) {
+    if (searchText == "") {
+      searchedValues = [];
+    } else {
+      topListQuery = useTopListQuery(config, metricsDefId, activeDimensionId, {
```

Comment:
This is a bit counter-intuitive that we're using the TopList API w/o any measures in order to fetch the dimension names. I get that it's because we don't currently have a more explicit endpoint to do this job, but it'd help to see a code comment explain the situation.